### PR TITLE
Skip cocoon backend cloud build if no new commits

### PR DESCRIPTION
**Pull request by keyonghan**
This is for https://github.com/flutter/flutter/issues/87749.

### CODE REVIEW

File: cloudbuild.yaml [hunk 1]
```diff
@@ -1,21 +1,61 @@
...
+        else
+          bash cloud_build/repo_dash_build.sh
+        fi
 
   # Deploy a new version to google cloud.
   - name: 'gcr.io/cloud-builders/gcloud'
     entrypoint: '/bin/bash'
-    args: ['cloud_build/deploy.sh', '$PROJECT_ID', '$SHORT_SHA', '$_GAE_PROMOTE']
+    args:  
+      - '-c'
```
Comment (CaseyHillers) [inline cloudbuild.yaml:52]: Couldn't we just add the check in the deploy step? e.g. build everything, but only deploy if it's a new version?

Comment (keyonghan) [reply cloudbuild.yaml:52]: I don't see the value to build everything we have already built/deployed before.

Comment (CaseyHillers) [reply cloudbuild.yaml:52]: Cocoon changes aren't the only thing that break our deploys, and it's helpful to know when those start to occur

File: cloudbuild.yaml [hunk 2]
```diff
@@ -1,21 +1,61 @@
...
+        echo "latest_version: $latest_version"
+        echo "current_version: $current_version"
+        if [ "$latest_version" = "$current_version" ]; then
+          echo "No updates since last deployment."
+        else
+          bash cloud_build/app_flutter_build.sh
+        fi
 
-    # Build repo_dashboard.
+  #   # Build repo_dashboard.
```
Comment (CaseyHillers) [inline cloudbuild.yaml:35]: nit: extra #

File: cloudbuild.yaml (hunk 1 above)
Comment (keyonghan) [reply cloudbuild.yaml:22]: Ah, missed that part. Updated to always build.

File: cloudbuild.yaml (hunk 2 above)
Comment (keyonghan) [reply cloudbuild.yaml]: Done.

**Review by CaseyHillers — APPROVED** [review]

**Pull request merged**